src/ontology/temp_mondo_gard_obsoletes.py

```diff
@@ -0,0 +1,15 @@
+"""GARD::Mondo obsoletes"""
```

Comment:
@matentzn This is a temporary script until I get clarity on [some questions](https://github.com/monarch-initiative/mondo/issues/6436#issuecomment-1637254018) I had in the issue (#6436).

Using some makefile commands to create `tmp/obsoletes.tsv` and `tmp/mondo-edit.sssom.tsv` I got the following table, but it is only 55 entries instead of the expected 785 (maybe because we need to use my new GARD mappings, not the ones already here).

[gard-mondo-mapped-obsoletes.tsv.zip](https://github.com/monarch-initiative/mondo/files/12065540/gard-mondo-mapped-obsoletes.tsv.zip)

|subject_id   |subject_label                                                                      |predicate_id |object_id   |object_label|mapping_justification     |subject_source|object_source|
|-------------|-----------------------------------------------------------------------------------|-------------|------------|------------|--------------------------|--------------|-------------|
|MONDO:0000501|obsolete Jensen syndrome                                                           |oio:hasDbXref|GARD:0003046|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0008068|obsolete navicular bone, accessory                                                 |oio:hasDbXref|GARD:0008543|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0008351|obsolete pupil, egg-shaped                                                         |oio:hasDbXref|GARD:0008291|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0008761|obsolete alpha-2-deficient collagen disease                                        |oio:hasDbXref|GARD:0000616|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0008761|obsolete alpha-2-deficient collagen disease                                        |oio:hasDbXref|GARD:0003454|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0009428|obsolete childhood hypophosphatasia                                                |oio:hasDbXref|GARD:0008735|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0009930|obsolete pulmonary arteriovenous malformation                                      |oio:hasDbXref|GARD:0004584|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0010045|obsolete paraplegia-brachydactyly-cone-shaped epiphysis syndrome                   |oio:hasDbXref|GARD:0002343|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0010086|obsolete sudden infant death syndrome                                              |oio:hasDbXref|GARD:0007711|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0010994|obsolete micromelic dwarfism, Fryns type                                           |oio:hasDbXref|GARD:0003642|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0011910|obsolete autosomal dominant limb-girdle muscular dystrophy type 1C                 |oio:hasDbXref|GARD:0012527|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0012294|obsolete drug metabolism, poor, Cyp2C19-related                                    |oio:hasDbXref|GARD:0012906|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0012501|obsolete mutagen sensitivity                                                       |oio:hasDbXref|GARD:0008273|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0015108|obsolete rare non-syndromic intellectual disability                                |oio:hasDbXref|GARD:0012633|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0016250|obsolete rare adenocarcinoma of the breast                                         |oio:hasDbXref|GARD:0012773|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0016766|obsolete rare lichen planus                                                        |oio:hasDbXref|GARD:0012344|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0017458|obsolete postaxial polydactyly of toes                                             |oio:hasDbXref|GARD:0012770|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0017753|obsolete disorder of peroxisomal alpha-, beta- and omega-oxidation                 |oio:hasDbXref|GARD:0012476|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0017984|obsolete familial lambdoid synostosis                                              |oio:hasDbXref|GARD:0003168|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0018132|obsolete congenital muscular alpha-dystroglycanopathy with brain and eye anomalies |oio:hasDbXref|GARD:0012588|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0018296|obsolete congenital disorder of glycosylation with developmental anomaly           |oio:hasDbXref|GARD:0012782|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0018391|obsolete male infertility with spermatogenesis disorder                            |oio:hasDbXref|GARD:0012512|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0018392|obsolete male infertility with spermatogenesis disorder due to single gene mutation|oio:hasDbXref|GARD:0012513|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0018598|obsolete neonatal adrenoleukodystrophy                                             |oio:hasDbXref|GARD:0000559|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0019174|obsolete infantile Refsum disease                                                  |oio:hasDbXref|GARD:0004648|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0019779|obsolete Renier-Gabreels-Jasper syndrome                                           |oio:hasDbXref|GARD:0004672|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0020050|obsolete autosomal trisomy                                                         |oio:hasDbXref|GARD:0006065|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0020078|obsolete acute myeloid leukemia with recurrent genetic anomaly                     |oio:hasDbXref|GARD:0012758|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0020668|obsolete spastic paraplegia 5B                                                     |oio:hasDbXref|GARD:0004917|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0020670|obsolete antithrombin deficiency type 2                                            |oio:hasDbXref|GARD:0010182|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0020742|obsolete cataract, microcephaly, failure to thrive, kyphoscoliosis syndrome        |oio:hasDbXref|GARD:0001060|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0021170|obsolete amyotonia congenita                                                       |oio:hasDbXref|GARD:0005798|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0021849|obsolete alopecia macular degeneration growth retardation syndrome                 |oio:hasDbXref|GARD:0000610|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0022109|obsolete catatrichy                                                                |oio:hasDbXref|GARD:0010080|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0022412|obsolete albinism immunodeficiency                                                 |oio:hasDbXref|GARD:0000590|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0022448|obsolete amyoplasia mandibulofacial dysostosis                                     |oio:hasDbXref|GARD:0000660|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0022464|obsolete anophthalmia microcephaly hypogonadism                                    |oio:hasDbXref|GARD:0000718|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0022482|obsolete apolipoprotein C 2i deficiency                                            |oio:hasDbXref|GARD:0000759|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0022495|obsolete arthritis short stature deafness                                          |oio:hasDbXref|GARD:0000775|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0022732|obsolete chorea minor                                                              |oio:hasDbXref|GARD:0006057|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0022794|obsolete chromosome 8 deletion                                                     |oio:hasDbXref|GARD:0012070|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0022916|obsolete cystic hygroma lethal cleft palate                                        |oio:hasDbXref|GARD:0001653|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0022918|obsolete cytokine deficiency                                                       |oio:hasDbXref|GARD:0009529|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0022921|obsolete short stature-microcephaly-heart defect syndrome                          |oio:hasDbXref|GARD:0000233|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0023072|obsolete envenomization by Bothrops lanceolatus                                    |oio:hasDbXref|GARD:0002131|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0023097|obsolete exostoses anetodermia brachydactyly type E                                |oio:hasDbXref|GARD:0002202|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0023141|obsolete antihypertensive drugs antenatal exposure syndrome                        |oio:hasDbXref|GARD:0000733|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0023156|obsolete fibular aplasia                                                           |oio:hasDbXref|GARD:0008659|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0023174|obsolete follicular lymphoreticuloma                                               |oio:hasDbXref|GARD:0002357|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0023663|obsolete macrocephaly mesodermal hamartoma spectrum                                |oio:hasDbXref|GARD:0000170|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0023807|obsolete midphalangeal hair                                                        |oio:hasDbXref|GARD:0009992|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0023873|obsolete Noonan-like/multiple giant cell lesion syndrome                           |oio:hasDbXref|GARD:0004006|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0024996|obsolete Usher syndrome, type 2b                                                   |oio:hasDbXref|GARD:0005441|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0027416|obsolete retinal cone dystrophy 2                                                  |oio:hasDbXref|GARD:0010117|            |semapv:UnspecifiedMatching|MONDO         |GARD         |
|MONDO:0100229|obsolete Heimler syndrome                                                          |oio:hasDbXref|GARD:0001687|            |semapv:UnspecifiedMatching|MONDO         |GARD         |

Reply:
Interestingly enough, I also ran `scripts/deprecated_in_mondo.py` in `mondo-ingest`, and got 0 results. Hmmm.. I suppose, now that I think of it, that is because it is supposed to answer the question of "Deprecated / obsolete terms w/ have xrefs in Mondo which have not yet been marked as deprecated in Mondo" as opposed to "terms in the target ontology that are deprecated in Mondo".

Reply:
What is this script supposed to do? I mean, high level? 

Can you commit the oak output for the obsoleted command, then we iterate on that. :) merci

Reply:
Yeah, I put `temp_` in the name because I'm not sure if this is really the implementation we want. It's a very short script. Basically I used the OAK commands `obsoletes` and `sssom` to get all the information I needed to answer our question. But I wasn't able to figure out how to do it in 100% OAK, without needing to leverage unix or Python etc to use both of these tables as inputs to generate the final output you see above.

Reply:
@matentzn Note that I read through the whole OAK tutorial / guide in pretty good detail as Chris suggested, but I did not find a way to do this with only OAK commands (not sure if that's what you want; I guess only if it's actually possible).

I did get possibly closer to that though. I was able to get this chaining to work successfully:
```make
tmp/mondo-obsoletes.sssom.tsv:
	runoak -i sqlite:tmp/mondo.db obsoletes | runoak -i sqlite:tmp/mondo.db mappings -O sssom -o $@ -
```

Result: [mondo-obsoletes.sssom.tsv.zip](https://github.com/monarch-initiative/mondo/files/12074758/mondo-obsoletes.sssom.tsv.zip)

I'm not sure if you want this committed to `reports/`. And if so, do you still want `reports/obsoletes.tsv`? I don't actually need that anymore.

Anyway all the script does now is takes `tmp/mondo-obsoletes.sssom.tsv` and filters it by "GARD".

The script is very ad-hoc and simple (doesn't take CLI params; doesn't have docstrings), because I'm not sure if you will like this implementation or not; so I didn't want to over-commit to it. 

If you approve of it, let me know and I will move it into `scripts/` and make it proper. Otherwise, let me know if you want me to implement this some other way.

Reply:
We decided to use SPARQL because did not seem to be a way to get "obsoletion reason" because custom fields not available with these OAK commands. Maybe if we passed the obsoletes as input to some other command instead of `mappings`.